File: .agent/skills/cnma-cap-backend/templates/node/srv/src/domain/__feature__/events/OnRead{{FeatureName}}Event.js
import cds from '@sap/cds';

/**
 * OnRead{{FeatureName}}Event - READ event handler for {{FeatureName}} domain.
 * SRP: Contains only READ logic, no business rules.
 */
module.exports = {
    execute: async (req) => {
        try {
            const userId = req.user?.id || 'anonymous';

            // Extract query conditions from incoming request
            let reqConditions = req.query?.SELECT?.where || [];

            // Predefined conditions (e.g., filter by user)
            const predefinedConditions = [
                { ref: ['createdBy'] }, '=', { val: userId }
            ];

            // Combine request conditions with predefined ones
            let combinedCondition;
            if (reqConditions.length > 0) {
                combinedCondition = ['(', ...reqConditions, ')', 'and', ...predefinedConditions];
            } else {
                combinedCondition = predefinedConditions;
            }

            // Execute query with combined conditions
            const query = cds.query;
            let selectQuery = cds
                .select('{{FeatureName}}')
                .columns('*')
                .where(combinedCondition);

            const results = await cds.run(selectQuery);
            return results;
        } catch (error) {
            req.error(500, 'Error reading {{FeatureName}}: ' + error.message);
        }
    }
};

File: .agent/skills/cnma-cap-backend/templates/java/srv/src/main/java/cnma/{{module_name}}/domain/__feature__/events/OnRead{{FeatureName}}Event.java
package cnma.{{module_name}}.domain.__feature__.events;

import com.sap.cds.services.RequestContext;
import com.sap.cds.ql.Select;
import com.sap.cds.Result;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

/**
 * OnRead{{FeatureName}}Event - READ event handler.
 * SRP: Only handles READ logic, delegates persistence to Repository.
 */
@Slf4j
@Component
@RequiredArgsConstructor
public class OnRead{{FeatureName}}Event {

    private final RequestContext requestContext;

    public Object execute(Object req) {
        // Extract user context
        String userId = getUserId(req);
        log.debug("READ {{FeatureName}} by user: {}", userId);

        // Extract query conditions from request
        // Combine with predefined conditions (e.g., filter by user)
        // Execute query and return results
        return null;
    }

    private String getUserId(Object req) {
        return "anonymous";
    }
}